### PR TITLE
✨ Add basic support for scanning windows disk drives via device conn.

### PR DESCRIPTION
**Pull request by preslavgerchev**
Add support for identifying the drive letter where the windows partition is located by only specifying the LUN (logical unit number) or the serial number of the disk.

```
./cnquery.exe shell device --serial-number 123455
```

```
./cnquery.exe shell device --lun 2
```

There's also support for setting a disk online if it's found to be offline. This only happens if we that this is the target disk that we're searching for. Once scanning is done, the disk is set back to offline.

### CODE REVIEW

**Review by afiune — COMMENTED** [review]
About to test this!

File: providers/os/connection/device/windows/device_manager.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	}
+	partitionInfo := &snapshot.PartitionInfo{
+		Name:   partition.DriveLetter,
+		FsType: "Windows",
+	}
+	return []*snapshot.PartitionInfo{partitionInfo}, nil
+}
+
+// validates the options provided to the device manager
+func validateOpts(opts map[string]string) error {
```
Comment (afiune) [inline providers/os/connection/device/windows/device_manager.go:82]: can we test this? What if these options doesn't exist inside the `map[]`?

File: providers/os/connection/device/windows/disk_drive.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		if err != nil {
+			return nil, err
+		}
+		return []*diskPartition{partition}, nil
+	}
+
+	return partitions, nil
+}
+
+func filterDiskDrives(drives []*diskDrive, opts map[string]string) (*diskDrive, error) {
```
Comment (afiune) [inline providers/os/connection/device/windows/disk_drive.go:156]: same here, this can be unit tested.

**Review by afiune — APPROVED** [review]

File: providers/os/connection/device/windows/device_manager.go (hunk 1 above)
Comment (preslavgerchev) [reply providers/os/connection/device/windows/device_manager.go:82]: great point, tests aded

**Pull request merged**